python/dgl/nn/pytorch/link.py

```diff
@@ -0,0 +1,170 @@
+"""Torch modules for link prediction."""
+# pylint: disable= no-member, arguments-differ, invalid-name, W0235
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+__all__ = ['EdgePredictor']
+
+class EdgePredictor(nn.Module):
+    r"""
+
+    Description
+    -----------
+    Predictor/score function for pairs of node representations. Given a pair of node
+    representations, :math:`h_i` and :math:`h_j`, it combines them with
+
+    **dot product**
+
+    .. math::
+
+        h_i^{T} h_j
+
+    or **cosine similarity**
+
+    .. math::
+
+        \frac{h_i^{T} h_j}{{\| h_i \|}_2 \cdot {\| h_j \|}_2}
+
+    or **elementwise product**
+
+    .. math::
+
+        h_i \odot h_j
+
+    or **concatenation**
+
+    .. math::
+
+        h_i \Vert h_j
+
+    Optionally, it passes the combined results to a linear layer for the final prediction.
+
+    Parameters
+    ----------
+    op : str
+        The operation to apply. It can be 'dot', 'cos', 'ele', or 'cat',
+        corresponding to the equations above in order.
+    in_feats : int, optional
+        The input feature size of :math:`h_i` and :math:`h_j`. It is required
+        only if a linear layer is to be applied.
+    out_feats : int, optional
+        The output feature size. It is reuiqred only if a linear layer is to be applied.
+    bias : bool, optional
+        Whether to use bias for the linear layer if it applies.
+
+    Examples
+    --------
+    >>> import torch as th
```

Comment:
It's better to have an example starting from a graph (how to slice data from edge to node).

Reply:
Done.